app/src/main/java/com/internship/move/ui/main/viewmodel/MainViewModel.kt

```diff
@@ -0,0 +1,58 @@
+package com.internship.move.ui.main.viewmodel
+
+import android.content.Context.MODE_PRIVATE
+import androidx.lifecycle.ViewModel
+import com.internship.move.MainActivity
+import com.internship.move.util.Constants.SharedPref.KEY_APP_PREFERENCES
+import com.internship.move.util.Constants.SharedPref.KEY_HAS_VISITED_AUTHENTICATION
+import com.internship.move.util.Constants.SharedPref.KEY_HAS_VISITED_ONBOARDING
+
+class MainViewModel : ViewModel() {
```

Comment:
I see you're using a single viewModel that does everything.
I would suggest breaking it apart into smaller vMs where each has only the methods it needs to work instead of using this god class that handles everything.
This way we make sure changes aren't made from places they aren't supposed to be made (separation of concerns).